### ISSUE TITLE
About the bboxes.

### ISSUE THREAD

**Issue by jhl13** [ZJULearning/pixel_link#85]
In the file mtwi_to_tfrecords.py, line 123.
why bboxes.append([xmin, ymin, xmax, ymax])?
I think it is bboxes.append([ymin, xmin, ymax, xmax]), because in tf.image.draw_bounding_boxes, The coordinates of the each bounding box in boxes are encoded as [y_min, x_min, y_max, x_max].
so in ssd_vgg_preprocessing.py, tf.image.sample_distorted_bounding_box return the distort_bboxes encoded as [y_min, x_min, y_max, x_max].

**Comment by northeastsquare** [comment]
在dataset_utils.py里面，做了对应
    example = tf.train.Example(features=tf.train.Features(feature={
            'image/shape': int64_feature(list(shape)),
            'image/object/bbox/xmin': float_feature(get_list(bboxes, 0)),
            'image/object/bbox/ymin': float_feature(get_list(bboxes, 1)),
            'image/object/bbox/xmax': float_feature(get_list(bboxes, 2)),
            'image/object/bbox/ymax': float_feature(get_list(bboxes, 3)),
            'image/object/bbox/x1': float_feature(get_list(oriented_bboxes, 0)),
            'image/object/bbox/y1': float_feature(get_list(oriented_bboxes, 1)),
            'image/object/bbox/x2': float_feature(get_list(oriented_bboxes, 2)),
            'image/object/bbox/y2': float_feature(get_list(oriented_bboxes, 3)),
            'image/object/bbox/x3': float_feature(get_list(oriented_bboxes, 4)),
            'image/object/bbox/y3': float_feature(get_list(oriented_bboxes, 5)),
            'image/object/bbox/x4': float_feature(get_list(oriented_bboxes, 6)),
            'image/object/bbox/y4': float_feature(get_list(oriented_bboxes, 7)),
            'image/object/bbox/label': int64_feature(labels),
            'image/object/bbox/label_text': bytes_feature(labels_text),
            'image/format': bytes_feature(image_format),
            'image/filename': bytes_feature(filename),
            'image/encoded': bytes_feature(image_data)}))

**Comment by jhl13** [comment]
> 在dataset_utils.py里面，做了对应
> example = tf.train.Example(features=tf.train.Features(feature={
> 'image/shape': int64_feature(list(shape)),
> 'image/object/bbox/xmin': float_feature(get_list(bboxes, 0)),
> 'image/object/bbox/ymin': float_feature(get_list(bboxes, 1)),
> 'image/object/bbox/xmax': float_feature(get_list(bboxes, 2)),
> 'image/object/bbox/ymax': float_feature(get_list(bboxes, 3)),
> 'image/object/bbox/x1': float_feature(get_list(oriented_bboxes, 0)),
> 'image/object/bbox/y1': float_feature(get_list(oriented_bboxes, 1)),
> 'image/object/bbox/x2': float_feature(get_list(oriented_bboxes, 2)),
> 'image/object/bbox/y2': float_feature(get_list(oriented_bboxes, 3)),
> 'image/object/bbox/x3': float_feature(get_list(oriented_bboxes, 4)),
> 'image/object/bbox/y3': float_feature(get_list(oriented_bboxes, 5)),
> 'image/object/bbox/x4': float_feature(get_list(oriented_bboxes, 6)),
> 'image/object/bbox/y4': float_feature(get_list(oriented_bboxes, 7)),
> 'image/object/bbox/label': int64_feature(labels),
> 'image/object/bbox/label_text': bytes_feature(labels_text),
> 'image/format': bytes_feature(image_format),
> 'image/filename': bytes_feature(filename),
> 'image/encoded': bytes_feature(image_data)}))

yep
but i think bboxes[0] is correspond to ymin, not xmin.

**Comment by rkshuai** [comment]
In the get_split function, it has 
 'object/bbox': slim.tfexample_decoder.BoundingBox(
   ['ymin', 'xmin', 'ymax', 'xmax'], 'image/object/bbox/'),

**Comment by northeastsquare** [comment]
上面老兄说的对，
train_pixel_link.py里面，取object/bbox,就是rkshuai说的这句，
‘object/bbox’,顺序是ymin, xmin, ymax,xmax
`[image, glabel, gbboxes, x1, x2, x3, x4, y1, y2, y3, y4] = provider.get([
                                                         'image',
                                                         'object/label',
                                                         'object/bbox', 
                                                         'object/oriented_bbox/x1',
                                                         'object/oriented_bbox/x2',
                                                         'object/oriented_bbox/x3',
                                                         'object/oriented_bbox/x4',
                                                         'object/oriented_bbox/y1',
                                                         'object/oriented_bbox/y2',
                                                         'object/oriented_bbox/y3',
                                                         'object/oriented_bbox/y4'
                                                         ])`

**Comment by jhl13** [comment]
> In the get_split function, it has
> 'object/bbox': slim.tfexample_decoder.BoundingBox(
> ['ymin', 'xmin', 'ymax', 'xmax'], 'image/object/bbox/'),

Thanks! i get it!
